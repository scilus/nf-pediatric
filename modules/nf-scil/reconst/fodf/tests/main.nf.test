nextflow_process {

    name "Test Process RECONST_FODF"
    script "../main.nf"
    process "RECONST_FODF"

    tag "modules"
    tag "modules_nfcore"
    tag "reconst"
    tag "reconst/fodf"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "processing.zip" , "commit_amico.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("reconst - fodf") {

        config "./nextflow.config"

        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        ssst: it.simpleName == "processing"
                        msmt: it.simpleName == "commit_amico"
                    }
                input[0] = ch_split_test_data.ssst.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi.bval"),
                        file("\${test_data_directory}/dwi.bvec"),
                        file("\${test_data_directory}/cc.nii.gz"),
                        file("\${test_data_directory}/fa.nii.gz"),
                        file("\${test_data_directory}/md.nii.gz"),
                        file("\${test_data_directory}/frf.txt"),
                        [],
                        []
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("reconst - fodf_with_shells") {

        config "./nextflow_shells.config"

        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        ssst: it.simpleName == "processing"
                        msmt: it.simpleName == "commit_amico"
                    }
                input[0] = ch_split_test_data.ssst.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi.bval"),
                        file("\${test_data_directory}/dwi.bvec"),
                        file("\${test_data_directory}/cc.nii.gz"),
                        file("\${test_data_directory}/fa.nii.gz"),
                        file("\${test_data_directory}/md.nii.gz"),
                        file("\${test_data_directory}/frf.txt"),
                        [],
                        []
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("reconst - fodf_no_mask") {

        config "./nextflow.config"

        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        ssst: it.simpleName == "processing"
                        msmt: it.simpleName == "commit_amico"
                    }
                input[0] = ch_split_test_data.ssst.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi.bval"),
                        file("\${test_data_directory}/dwi.bvec"),
                        [],
                        file("\${test_data_directory}/fa.nii.gz"),
                        file("\${test_data_directory}/md.nii.gz"),
                        file("\${test_data_directory}/frf.txt"),
                        [],
                        []
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("reconst - fodf_no_metrics") {

        config "./nextflow_no_metrics.config"

        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        ssst: it.simpleName == "processing"
                        msmt: it.simpleName == "commit_amico"
                    }
                input[0] = ch_split_test_data.ssst.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi.bval"),
                        file("\${test_data_directory}/dwi.bvec"),
                        file("\${test_data_directory}/cc.nii.gz"),
                        file("\${test_data_directory}/fa.nii.gz"),
                        file("\${test_data_directory}/md.nii.gz"),
                        file("\${test_data_directory}/frf.txt"),
                        [],
                        []
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("reconst - fodf_msmt") {

        config "./nextflow_msmt.config"

        when {
            process {
                """
                ch_split_test_data = LOAD_DATA.out.test_data_directory
                    .branch{
                        ssst: it.simpleName == "processing"
                        msmt: it.simpleName == "commit_amico"
                    }
                input[0] = ch_split_test_data.msmt.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ],
                        file("\${test_data_directory}/dwi.nii.gz"),
                        file("\${test_data_directory}/dwi.bval"),
                        file("\${test_data_directory}/dwi.bvec"),
                        file("\${test_data_directory}/mask.nii.gz"),
                        file("\${test_data_directory}/fa.nii.gz"),
                        file("\${test_data_directory}/md.nii.gz"),
                        file("\${test_data_directory}/wm_frf.txt"),
                        file("\${test_data_directory}/gm_frf.txt"),
                        file("\${test_data_directory}/csf_frf.txt")

                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.wm_fodf.get(0).get(1)).name,
                    file(process.out.gm_fodf.get(0).get(1)).name,
                    file(process.out.csf_fodf.get(0).get(1)).name,
                    file(process.out.vf.get(0).get(1)).name,
                    file(process.out.vf_rgb.get(0).get(1)).name,
                    file(process.out.peaks.get(0).get(1)).name,
                    file(process.out.peak_values.get(0).get(1)).name,
                    file(process.out.peak_indices.get(0).get(1)).name,
                    file(process.out.afd_max.get(0).get(1)).name,
                    file(process.out.afd_total.get(0).get(1)).name,
                    file(process.out.afd_sum.get(0).get(1)).name,
                    file(process.out.nufo.get(0).get(1)).name).match() }
            )
        }

    }

}
