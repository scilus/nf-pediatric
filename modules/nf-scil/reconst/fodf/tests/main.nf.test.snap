{
    "reconst - fodf_msmt": {
        "content": [
            "test__wm_fodf.nii.gz",
            "test__gm_fodf.nii.gz",
            "test__csf_fodf.nii.gz",
            "test__vf.nii.gz",
            "test__vf_rgb.nii.gz",
            "test__peaks.nii.gz",
            "test__peak_values.nii.gz",
            "test__peak_indices.nii.gz",
            "test__afd_max.nii.gz",
            "test__afd_total.nii.gz",
            "test__afd_sum.nii.gz",
            "test__nufo.nii.gz"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:04:27.988184"
    },
    "reconst - fodf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__fodf.nii.gz:md5,c45f0391a462d358930809502dd2da15"
                    ]
                ],
                "1": [
                    
                ],
                "10": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_total.nii.gz:md5,9bac465d58b8662c3dce2c4937300a5a"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_sum.nii.gz:md5,a849e3eb8c67cc176e88daca01669eef"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__nufo.nii.gz:md5,711f0ee8d01e71166099ffdf4c684b5f"
                    ]
                ],
                "13": [
                    
                ],
                "14": [
                    "versions.yml:md5,8f5a73278d63aac3bf8073c5d0ef488b"
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peaks.nii.gz:md5,fbc8af94bce991a6ebe3bfb837d39a52"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_values.nii.gz:md5,4fd4a44c18544413c0e9bb672e60f709"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_indices.nii.gz:md5,d39ad843a7b86ac7042c44b87356e68b"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_max.nii.gz:md5,96b7a2dbab6306ca053b45bfad21be55"
                    ]
                ],
                "afd_max": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_max.nii.gz:md5,96b7a2dbab6306ca053b45bfad21be55"
                    ]
                ],
                "afd_sum": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_sum.nii.gz:md5,a849e3eb8c67cc176e88daca01669eef"
                    ]
                ],
                "afd_total": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_total.nii.gz:md5,9bac465d58b8662c3dce2c4937300a5a"
                    ]
                ],
                "csf_fodf": [
                    
                ],
                "fodf": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__fodf.nii.gz:md5,c45f0391a462d358930809502dd2da15"
                    ]
                ],
                "gm_fodf": [
                    
                ],
                "nufo": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__nufo.nii.gz:md5,711f0ee8d01e71166099ffdf4c684b5f"
                    ]
                ],
                "peak_indices": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_indices.nii.gz:md5,d39ad843a7b86ac7042c44b87356e68b"
                    ]
                ],
                "peak_values": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_values.nii.gz:md5,4fd4a44c18544413c0e9bb672e60f709"
                    ]
                ],
                "peaks": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peaks.nii.gz:md5,fbc8af94bce991a6ebe3bfb837d39a52"
                    ]
                ],
                "vent_mask": [
                    
                ],
                "versions": [
                    "versions.yml:md5,8f5a73278d63aac3bf8073c5d0ef488b"
                ],
                "vf": [
                    
                ],
                "vf_rgb": [
                    
                ],
                "wm_fodf": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-01T19:21:03.487368"
    },
    "reconst - fodf_no_metrics": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__fodf.nii.gz:md5,c45f0391a462d358930809502dd2da15"
                    ]
                ],
                "1": [
                    
                ],
                "10": [
                    
                ],
                "11": [
                    
                ],
                "12": [
                    
                ],
                "13": [
                    
                ],
                "14": [
                    "versions.yml:md5,8f5a73278d63aac3bf8073c5d0ef488b"
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    
                ],
                "afd_max": [
                    
                ],
                "afd_sum": [
                    
                ],
                "afd_total": [
                    
                ],
                "csf_fodf": [
                    
                ],
                "fodf": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__fodf.nii.gz:md5,c45f0391a462d358930809502dd2da15"
                    ]
                ],
                "gm_fodf": [
                    
                ],
                "nufo": [
                    
                ],
                "peak_indices": [
                    
                ],
                "peak_values": [
                    
                ],
                "peaks": [
                    
                ],
                "vent_mask": [
                    
                ],
                "versions": [
                    "versions.yml:md5,8f5a73278d63aac3bf8073c5d0ef488b"
                ],
                "vf": [
                    
                ],
                "vf_rgb": [
                    
                ],
                "wm_fodf": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-01T20:22:29.171158"
    },
    "reconst - fodf_with_shells": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__fodf.nii.gz:md5,c45f0391a462d358930809502dd2da15"
                    ]
                ],
                "1": [
                    
                ],
                "10": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_total.nii.gz:md5,9bac465d58b8662c3dce2c4937300a5a"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_sum.nii.gz:md5,a849e3eb8c67cc176e88daca01669eef"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__nufo.nii.gz:md5,711f0ee8d01e71166099ffdf4c684b5f"
                    ]
                ],
                "13": [
                    
                ],
                "14": [
                    "versions.yml:md5,8f5a73278d63aac3bf8073c5d0ef488b"
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peaks.nii.gz:md5,fbc8af94bce991a6ebe3bfb837d39a52"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_values.nii.gz:md5,4fd4a44c18544413c0e9bb672e60f709"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_indices.nii.gz:md5,d39ad843a7b86ac7042c44b87356e68b"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_max.nii.gz:md5,96b7a2dbab6306ca053b45bfad21be55"
                    ]
                ],
                "afd_max": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_max.nii.gz:md5,96b7a2dbab6306ca053b45bfad21be55"
                    ]
                ],
                "afd_sum": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_sum.nii.gz:md5,a849e3eb8c67cc176e88daca01669eef"
                    ]
                ],
                "afd_total": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_total.nii.gz:md5,9bac465d58b8662c3dce2c4937300a5a"
                    ]
                ],
                "csf_fodf": [
                    
                ],
                "fodf": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__fodf.nii.gz:md5,c45f0391a462d358930809502dd2da15"
                    ]
                ],
                "gm_fodf": [
                    
                ],
                "nufo": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__nufo.nii.gz:md5,711f0ee8d01e71166099ffdf4c684b5f"
                    ]
                ],
                "peak_indices": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_indices.nii.gz:md5,d39ad843a7b86ac7042c44b87356e68b"
                    ]
                ],
                "peak_values": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_values.nii.gz:md5,4fd4a44c18544413c0e9bb672e60f709"
                    ]
                ],
                "peaks": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peaks.nii.gz:md5,fbc8af94bce991a6ebe3bfb837d39a52"
                    ]
                ],
                "vent_mask": [
                    
                ],
                "versions": [
                    "versions.yml:md5,8f5a73278d63aac3bf8073c5d0ef488b"
                ],
                "vf": [
                    
                ],
                "vf_rgb": [
                    
                ],
                "wm_fodf": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-01T20:38:04.916377"
    },
    "reconst - fodf_no_mask": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__fodf.nii.gz:md5,1c0aa14841e90acbdc048c6f0b117ce2"
                    ]
                ],
                "1": [
                    
                ],
                "10": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_total.nii.gz:md5,713fae8f03e15434034c21dc647f4a20"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_sum.nii.gz:md5,e907c21b54410ad69531b58b2c6a5e5d"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__nufo.nii.gz:md5,0e58d2571878aa13882f858cb4e572ad"
                    ]
                ],
                "13": [
                    
                ],
                "14": [
                    "versions.yml:md5,8f5a73278d63aac3bf8073c5d0ef488b"
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peaks.nii.gz:md5,3919c8117b3b800f52fc7319f523eb13"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_values.nii.gz:md5,e92f556fe1a926d088590dd2a7d970d6"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_indices.nii.gz:md5,ab4c1129a3ede2db638115e71258a288"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_max.nii.gz:md5,db6d8bd847e08e2857173f572b46670a"
                    ]
                ],
                "afd_max": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_max.nii.gz:md5,db6d8bd847e08e2857173f572b46670a"
                    ]
                ],
                "afd_sum": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_sum.nii.gz:md5,e907c21b54410ad69531b58b2c6a5e5d"
                    ]
                ],
                "afd_total": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__afd_total.nii.gz:md5,713fae8f03e15434034c21dc647f4a20"
                    ]
                ],
                "csf_fodf": [
                    
                ],
                "fodf": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__fodf.nii.gz:md5,1c0aa14841e90acbdc048c6f0b117ce2"
                    ]
                ],
                "gm_fodf": [
                    
                ],
                "nufo": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__nufo.nii.gz:md5,0e58d2571878aa13882f858cb4e572ad"
                    ]
                ],
                "peak_indices": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_indices.nii.gz:md5,ab4c1129a3ede2db638115e71258a288"
                    ]
                ],
                "peak_values": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peak_values.nii.gz:md5,e92f556fe1a926d088590dd2a7d970d6"
                    ]
                ],
                "peaks": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__peaks.nii.gz:md5,3919c8117b3b800f52fc7319f523eb13"
                    ]
                ],
                "vent_mask": [
                    
                ],
                "versions": [
                    "versions.yml:md5,8f5a73278d63aac3bf8073c5d0ef488b"
                ],
                "vf": [
                    
                ],
                "vf_rgb": [
                    
                ],
                "wm_fodf": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-01T19:50:32.073387"
    }
}